pkg/controllers/dashboard/plugin/controller.go

```diff
@@ -0,0 +1,75 @@
+package plugin
+
+import (
+	"context"
+	"fmt"
+
+	v1 "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	plugincontroller "github.com/rancher/rancher/pkg/generated/controllers/catalog.cattle.io/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func Register(
+	ctx context.Context,
+	systemNamespace string,
+	plugin plugincontroller.UIPluginController,
+	pluginCache plugincontroller.UIPluginCache,
+) {
+	h := &handler{
+		systemNamespace: systemNamespace,
+		plugin:          plugin,
+		pluginCache:     pluginCache,
+	}
+	plugin.OnChange(ctx, "on-plugin-change", h.OnPluginChange)
```

Comment:
nit: `on-plugin-change` to `on-ui-plugin-change`

Reply:
Done